src/Language/Tokenizer/Finder/TokenFinder.php

```diff
@@ -72,8 +72,29 @@ public function __invoke(Iterator $characters): TokenInterface
             throw new UnexpectedTokenSequenceException($sequence);
         }
 
-        $strategy = reset($previous);
+        $sequence = substr($sequence, 0, -1);
```

Comment:
Excellent observation. However, the strategies only accept a string and strict types are not enabled, so it will implicitly be cast to an empty string again. In reality that state is also never reached, but it's good to know that the code won't break. Your example is a really good reason to start covering the code with unit tests at some point, because then your scenario could be tested. I may start at least covering the part that is concerned with lexing, because that's not touching code that is likely to change significantly. But I would consider that out of scope.